test/integration/parachain/staging/sequential/issue.test.ts

```diff
@@ -221,44 +230,46 @@ describe("issue", () => {
-                assert.isTrue(issueRequest.status === IssueStatus.Cancelled, "Failed to cancel issue request");
-
-                // Set issue period back to its initial value to minimize side effects.
-                await sudo(userInterBtcAPI, () => userInterBtcAPI.issue.setIssuePeriod(initialIssuePeriod));
-
-            } catch (e) {
-                // Set issue period back to its initial value to minimize side effects.
-                await sudo(userInterBtcAPI, () => userInterBtcAPI.issue.setIssuePeriod(initialIssuePeriod));
-                throw e;
-            }
-        });
-    }).timeout(5 * 60000);
+        for (const vault_2_id of vault_2_ids) {
+            await runWhileMiningBTCBlocks(bitcoinCoreClient, async () => {
+                const initialIssuePeriod = await userInterBtcAPI.issue.getIssuePeriod();
+                await sudo(userInterBtcAPI, () => userInterBtcAPI.issue.setIssuePeriod(0));
+                try {
+                    // request issue
+                    const amount = newMonetaryAmount(0.0000121, wrappedCurrency, true);
+                    const vaultCollateralIdLiteral = currencyIdToLiteral(vault_2_id.currencies.collateral);
+                    const requestResults = await userInterBtcAPI.issue.request(
+                        amount, 
+                        newAccountId(api, vault_2.address), 
+                        vaultCollateralIdLiteral
+                    );
+                    assert.equal(requestResults.length, 1, "Test broken: more than one issue request created"); // sanity check
+                    const requestResult = requestResults[0];
+    
+                    // Wait for issue expiry callback
+                    await new Promise<void>((resolve, _) => {
+                        // userInterBtcAPI.issue.subscribeToIssueExpiry(newAccountId(api, userAccount.address), (requestId) => {
+                        //     if (stripHexPrefix(requestResult.id.toString()) === stripHexPrefix(requestId.toString())) {
+                        //         resolve();
+                        //     }
+                        // });
+                    });
+    
+                    await userInterBtcAPI.issue.cancel(requestResult.id);
+    
+                    const issueRequest = await userInterBtcAPI.issue.getRequestById(requestResult.id);
+                    assert.isTrue(issueRequest.status === IssueStatus.Cancelled, "Failed to cancel issue request");
+    
+                    // Set issue period back to its initial value to minimize side effects.
+                    await sudo(userInterBtcAPI, () => userInterBtcAPI.issue.setIssuePeriod(initialIssuePeriod));
+    
+                } catch (e) {
+                    // Set issue period back to its initial value to minimize side effects.
+                    await sudo(userInterBtcAPI, () => userInterBtcAPI.issue.setIssuePeriod(initialIssuePeriod));
+                    throw e;
+                }
```

Comment:
Good idea to reset the state, though I wonder if we should have some common routine that runs after most tests to "re-genesis" from a blank state. However I guess that may dramatically slowdown the pipeline if Substrate doesn't have an easy way to do that though.

Reply:
Once we have instant seal, we might be able to afford re-genesis delays, assuming it can be done.
I'm certainly open to the idea.